### PR TITLE
fix: Breaking change in MachineAuthProvider constructor

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
A change introduced in https://github.com/apache/superset/pull/25247 in MachineAuthProvider caused some custom auth providers to be broken, because of unexpected second argument passed to the constructor. This PR reverts that behaviour and .allows only passing either Selenium or Playwright override auth function, not both at the same time.

### TESTING INSTRUCTIONS
1. Add a custom `MACHINE_AUTH_PROVIDER_CLASS` that inherits from `MachineAuthProvider` and uses an `__init__` method that accepts only 1 argument - custom auth function (existing implementations follow this pattern)
2. Verify that your custom auth provider works as expected.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/utils/machine_auth.py [hunk 1]
```diff
@@ -43,15 +43,15 @@
 class MachineAuthProvider:
     def __init__(
         self,
-        auth_webdriver_func_override: Callable[[WebDriver, User], WebDriver],
-        auth_context_func_override: Callable[[BrowserContext, User], BrowserContext],
+        auth_webdriver_func_override: Callable[
+            [WebDriver | BrowserContext, User], WebDriver | BrowserContext
+        ],
```
Comment (villebro) [inline superset/utils/machine_auth.py:48]: The way we were using these was IMO incorrect - previously `auth_webdriver_func_override` was in fact optional, not always a `Callable`. So I propose changing the sig to make these optional, and defaulting them to `None`, like so:

```python
auth_webdriver_func_override: Callable[[WebDriver, User], WebDriver] | None = None,
```

**Review by villebro — COMMENTED** [review]
I propose we keep the original design, but update the sig to conform with how this is actually used, which assumes that the overrides are actually Optional arguments that default to `None`.

**Review by villebro — APPROVED** [review]
LGTM, but hoping we can clean this up properly in the next major version to have the same number of overrides as auth methods.

**Review by craig-rueda — APPROVED** [review]
LGTM, but would be great if there was a test or two for this :D

**Comment by john-bodley** [comment]
@kgabryje would you mind adding your testing instructions?

**Review by dpgaspar — APPROVED** [review]

**Comment by kgabryje** [comment]
Thanks for reviews!
@john-bodley Added test instructions. I had skipped this step as this PR is basically a revert to previous behaviour of the auth provider in order not to introduce a breaking change with the Playwright PR

**Pull request merged**